core/src/main/java/com/avast/bytes/Constants.java

```diff
@@ -2,4 +2,17 @@
 
 class Constants {
     static final char[] HEX_ARRAY = {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f'};
+
+    static int hexToBin(char ch) {
+        if ('0' <= ch && ch <= '9') {
+            return ch - '0';
+        }
+        if ('A' <= ch && ch <= 'F') {
+            return ch - 'A' + 10;
+        }
+        if ('a' <= ch && ch <= 'f') {
+            return ch - 'a' + 10;
+        }
+        return -1;
```

Comment:
I think the original implementation was copied from somewhere. Anyway, the method throws `IllegalArgumentExxception` in case there is bad input so I would do the same here too. And I don't like that it is a method that is part of class called `Constants` - that does not make sense.